lib/Extractor/KLEEBuilder.cpp

```diff
@@ -1101,6 +1101,9 @@ llvm::Optional<CandidateExpr> souper::GetCandidateExprForReplacement(
   // Build LHS
   ref<Expr> LHS = EB.get(Mapping.LHS);
   ref<Expr> Ante = klee::ConstantExpr::alloc(1, 1);
+  ref<Expr> DemandedBits = klee::ConstantExpr::alloc(Mapping.LHS->DemandedBitsVal);
+  if (!Mapping.LHS->DemandedBitsVal.isAllOnesValue())
+    Ante = AndExpr::create(Ante, EqExpr::create(AndExpr::create(LHS, DemandedBits), DemandedBits));
```

Comment:
I'm not convinced that this line is correct

Reply:
it looks like this is adding an antecedent forcing some bits to be zero, but what we really want to do is express that we don't care about the values of some of our output bits

Reply:
By masking, we say that nothing matters, right? That's why I did mask the demanded bits with LHS and expect the demanded bits in return.

Reply:
Oh wait, I think when I say that I expect demanded bits in return after masking, that's where I try to force zeros.

Reply:
see the new test cases I sent. those need to work, but they don't work now.

Reply:
Yes, working on all the review comments.